server/src/main/java/org/opensearch/action/bulk/BackoffPolicy.java

```diff
@@ -270,6 +281,48 @@ public TimeValue next() {
         }
     }
 
+    private static class ExponentialRandomBackoff extends BackoffPolicy {
+        private final long baseDelay;
+
+        private ExponentialRandomBackoff(long baseDelay) {
+            this.baseDelay = baseDelay;
+        }
+
+        @Override
+        public Iterator<TimeValue> iterator() {
+            return new ExponentialRandomBackoffIterator(baseDelay);
+        }
+    }
+
+    private static class ExponentialRandomBackoffIterator implements Iterator<TimeValue> {
+        /**
+         * Current delay in exponential backoff
+         */
+        private long currentDelay;
+
+        private ExponentialRandomBackoffIterator(long baseDelay) {
+            this.currentDelay = baseDelay;
+        }
+
+        /**
+         * There is not any limit for this BackOff.
+         * This Iterator will always return back off delay.
+         *
+         * @return true
+         */
+        @Override
+        public boolean hasNext() {
+            return true;
+        }
+
+        @Override
+        public TimeValue next() {
+            TimeValue delayToReturn = TimeValue.timeValueMillis(Randomness.get().nextInt(Math.toIntExact(currentDelay)) + 1);
+            currentDelay = Math.min(2 * currentDelay, Integer.MAX_VALUE);
```

Comment:
We want to first calculate the randomDelay out of current delay which we needs to return as part of current call and then double it and store in currentDelay for next call. 
So that's why we are first doing retrun delay calculation and then update current delay with double of it.